main.sh

```diff
@@ -16,4 +16,106 @@
 #
 # Copyright 2023 The Kepler Contributors
 #
-./common.sh $1
+
+set -eu -o pipefail
+
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+declare -r PROJECT_ROOT
+declare -r BIN_DIR="$PROJECT_ROOT/tmp/bin"
+
+source "$PROJECT_ROOT/.env"
```

Comment:
What's the purpose of using `.env` here?

Reply:
.env (which is ignored in `.gitignore` ) is a common way to define all env vars in a file that get precedence over those defined in the script. 
E.g. you can set your `CLUSTER_PROVIDER=microshift` in that file and `CTR_CMD=podman`

And since it is ignored in git, that change won't be pushed to upstream repos by mistake.

Reply:
1. where defined the default value?
2. what are they?
3. if we want to flexible how to make it?

Reply:
I have added a note on the usage.